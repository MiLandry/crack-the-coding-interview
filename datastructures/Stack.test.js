const Stack = require('./Stack');

let stack

beforeEach(() => {

stack = new Stack()
});

test('it adds', () => {
  stack.push('a')
  stack.push('b')
  stack.push('c')
  stack.push('d')
  expect(stack)
    .toMatchSnapshot()

})

test('it pops', () => {
  stack.push('a')
  stack.push('b')
  stack.push('c')
  stack.push('d')
  const topPlate = stack.pop()
  expect(topPlate)
    .toEqual('d')
  expect(stack)
    .toMatchSnapshot()

})
