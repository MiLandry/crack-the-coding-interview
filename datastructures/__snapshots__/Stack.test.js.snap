// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it adds 1`] = `
Stack {
  "data": Array [
    "a",
    "b",
    "c",
    "d",
  ],
}
`;

exports[`it pops 1`] = `
Stack {
  "data": Array [
    "a",
    "b",
    "c",
  ],
}
`;
